requests_kerberos/kerberos_.py

```diff
@@ -86,70 +84,75 @@ class HTTPKerberosAuth(AuthBase):
     object."""
     def __init__(
             self, mutual_authentication=REQUIRED,
-            service="HTTP", delegate=False):
+            service="HTTP", delegate=False, force_preemptive=False):
         self.context = {}
         self.mutual_authentication = mutual_authentication
         self.delegate = delegate
         self.pos = None
         self.service = service
+        self.force_preemptive = force_preemptive
 
-    def generate_request_header(self, response):
+    def generate_request_header(self, response, host, is_preemptive=False):
```

Comment:
Only if we want to kill the 401-based thing off. I'm not familiar enough with real-world Kerberos over HTTP to know, but the RFCs mention the possibility for multiple 401/Negotiate cycles before success. If we nuked this and just used the global setting, a (hypothetical) 401 resulting from a preemptive auth request would ignore the previous response and start a new exchange (probably not what we want). But I also have never seen a server that took more than one, so I don't know if that case even works here.